cuda/src/memory/cuda/resources/terminal/device_resource.cpp

```diff
@@ -0,0 +1,35 @@
+#include <memory>
+#include <memory_resource>
+
+#include <cuda_runtime.h>
+
+#include "vecmem/memory/cuda/resources/terminal/device_resource.hpp"
+
+namespace vecmem::memory::resources::terminal {
+    void * cuda_device_resource::do_allocate(
+        std::size_t bytes,
+        std::size_t
+    ) {
+        void * res;
+        cudaMalloc(&res, bytes);
+        return res;
+    }
+
+    void cuda_device_resource::do_deallocate(
+        void * p,
+        std::size_t,
+        std::size_t
+    ) {
+        cudaFree(p);
+    }
+
+    bool cuda_device_resource::do_is_equal(
+        const std::pmr::memory_resource & other
+    ) const noexcept {
+        return false;
```

Comment:
I would think that logically, whenever two resource objects are set up to talk to the same GPU, they should be considered equal, no?

Reply:
This was mostly a placeholder decision until we fleshed out the architecture for dealing with multiple devices a little more, but yes, in principle all resources dedicated to the same GPU can be considered equal (for now).